app/assets/javascripts/3dbio_viewer/src/webapp/components/model-upload/ModelUpload.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useCallback, useState, useRef } from "react";
+import { Dialog, DialogContent, DialogTitle, IconButton } from "@material-ui/core";
+import { Close } from "@material-ui/icons";
+import _ from "lodash";
+import i18n from "../../utils/i18n";
+import { Dropzone, DropzoneRef, getFile } from "../dropzone/Dropzone";
+import { useCallbackEffect } from "../../hooks/use-callback-effect";
+import { useAppContext } from "../AppContext";
+import "./ModelUpload.css";
+
+export interface ModelUploadProps {
+    title: string;
+    onClose(): void;
+}
+
+export const ModelUpload: React.FC<ModelUploadProps> = React.memo(props => {
+    const { title, onClose } = props;
+    const { compositionRoot } = useAppContext();
+
+    const [jobTitle, setJobTitle] = useState<string>("");
+    const [error, setError] = useState<string>();
+
+    const structureFileRef = useRef<DropzoneRef>(null);
+    const annotationFileRef = useRef<DropzoneRef>(null);
+
+    const onSubmitHandler = useCallback(() => {
```

Comment:
[good practices comment] In the case of actions, is a good practice to give more declarative names. The prop is called `onSubmitHandler`, that's ok, it's a generic prop, but here we know what it does. For examples: `const search = `.